autosklearn/data/validation.py

```diff
@@ -273,7 +314,24 @@ def _validate_dataframe_features(
                 self.feature_encoder.fit(X)
 
         if self.feature_encoder:
-            X = self.feature_encoder.transform(X)
+            try:
+                X = self.feature_encoder.transform(X)
+            except ValueError as e:
+                if 'Found unknown categories' in e.args[0]:
+                    # Make the message more informative
+                    raise ValueError(
+                        "During fit, the input features contained categorical values in columns"
+                        "{}, with categories {} which were encoded by Auto-sklearn automatically."
+                        "Nevertheless, a new input contained new categories not seen during "
```

Comment:
Could you please add that this is a limitation of scikit-learn (add a link to the open issue) so that it's clear to the user that this is not Auto-sklearn's fault?

Reply:
Done in next push!